src/EditorFeatures/CSharpTest/RemoveUnusedMembers/RemoveUnusedMembersTests.cs

```diff
@@ -0,0 +1,1632 @@
+        public async Task PropertyIsUnused()
+        {
+            await TestInRegularAndScriptAsync(
+@"class MyClass
+{
+    private int [|P|] { get; set; }
+}",
+@"class MyClass
+{
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedMembers)]
+        public async Task IndexerIsUnused()
+        {
+            await TestInRegularAndScriptAsync(
+@"class MyClass
+{
+    private int [|this|][int x] { get { return 0; } set { } }
+}",
+@"class MyClass
+{
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedMembers)]
+        public async Task EventIsUnused()
+        {
+            await TestInRegularAndScriptAsync(
+@"class MyClass
+{
+    private event System.EventHandler [|e|];
+}",
+@"class MyClass
+{
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedMembers)]
+        public async Task EntryPointMethodNotFlagged()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class MyClass
+{
+    private static void [|Main|]() { }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedMembers)]
+        public async Task EntryPointMethodNotFlagged_02()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"using System.Threading.Tasks;
+
+class MyClass
+{
+    private static async Task [|Main|]() => await Task.CompletedTask;
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedMembers)]
+        public async Task EntryPointMethodNotFlagged_03()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"using System.Threading.Tasks;
+
+class MyClass
+{
+    private static async Task<int> [|Main|]() => await Task.FromResult(0);
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedMembers)]
+        public async Task FieldIsUnused_ReadOnly()
+        {
+            await TestInRegularAndScriptAsync(
+@"class MyClass
+{
+    private readonly int [|_goo|];
+}",
+@"class MyClass
+{
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedMembers)]
+        public async Task PropertyIsUnused_ReadOnly()
+        {
+            await TestInRegularAndScriptAsync(
+@"class MyClass
+{
+    private int [|P|] { get; }
+}",
+@"class MyClass
+{
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedMembers)]
+        public async Task EventIsUnused_ReadOnly()
+        {
+            await TestInRegularAndScriptAsync(
+@"class MyClass
+{
+    private readonly event System.EventHandler [|E|];
+}",
+@"class MyClass
+{
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedMembers)]
+        public async Task FieldIsUnused_Static()
+        {
+            await TestInRegularAndScriptAsync(
+@"class MyClass
+{
+    private static int [|_goo|];
```

Comment:
do we have tests for extern/abstract?

Reply:
No, also reminds me I wanted to add tests for interface implementations. I will add these.

---
In reply to: [213407464](https://github.com/dotnet/roslyn/pull/29511#discussion_r213407464) [](ancestors = 213407464)